tests/e2e/restart_test.go

```diff
@@ -20,97 +18,141 @@
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 )
 
-type PluginSuite struct {
+type RestartTestData struct {
+	config        *inputparams.TestConfig
 	volumeName    string
-	hostIP        string
 	containerName string
-	esxIP         string
 }
 
-func (s *PluginSuite) SetUpTest(c *C) {
-	s.hostIP = os.Getenv("VM2")
-	s.esxIP = os.Getenv("ESX")
+func (s *RestartTestData) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping docker/plugin restart tests")
+	}
+
 	s.volumeName = inputparams.GetUniqueVolumeName("restart_test")
 	s.containerName = inputparams.GetUniqueContainerName("restart_test")
 
+	// ensure there are no remanants from an earlier run
+	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
+
 	// Create a volume
-	out, err := dockercli.CreateVolume(s.hostIP, s.volumeName)
+	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
-func (s *PluginSuite) TearDownTest(c *C) {
-	out, err := dockercli.RemoveContainer(s.hostIP, s.containerName)
-	c.Assert(err, IsNil, Commentf(out))
-	out, err = dockercli.DeleteVolume(s.hostIP, s.volumeName)
+func (s *RestartTestData) TearDownSuite(c *C) {
+	out, err := dockercli.DeleteVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
-var _ = Suite(&PluginSuite{})
+var _ = Suite(&RestartTestData{})
 
-// Test Stale mount
-// 1. Attach a volume to a container
+// TestVolumeDetached - verifies a volume is detached when the docker
+// daemon is stopped.
+// 1. Attach a volume to a container on a host
 // 2. Verify attached status
 // 3. Restart docker
 // 4. Verify detached status. Volume should be detached (within the timeout)
-func (s *PluginSuite) TestVolumeStaleMount(c *C) {
-	log.Printf("START: restart_test.TestVolumeStaleMount")
-
-	out, err := dockercli.AttachVolume(s.hostIP, s.volumeName, s.containerName)
+// 5. Verify a container can be started to use the same volume on another host
+// 6. Restart the docker daemon on the other host
+// 7. Verify detached status for the volume
+// 8. Verify a container can be started to use the same volume on the original host
+func (s *RestartTestData) TestVolumeDetached(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// 1. Attach a volume to a container on a host
+	out, err := dockercli.AttachVolume(s.config.DockerHosts[1], s.volumeName, s.containerName)
 	c.Assert(err, IsNil, Commentf(out))
 
-	status := verification.VerifyAttachedStatus(s.volumeName, s.hostIP, s.esxIP)
+	// 2. Verify attached status
+	status := verification.VerifyAttachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
 	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
 
-	out, err = dockercli.KillDocker(s.hostIP)
+	// 3. Restart docker
+	out, err = dockercli.KillDocker(s.config.DockerHosts[1])
```

Comment:
dockercli.KillDocker essentially kills docker. Docker daemon is expected to restart automatically. In case if it doesn't comeup automatically, systemctl is used  to bring docker up.
We can rename this util to RestartDocker if that avoids confusion.